src/hotspot/share/cds/aotMapLogger.hpp

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#ifndef SHARE_CDS_AOTMAPLOGGER_HPP
+#define SHARE_CDS_AOTMAPLOGGER_HPP
+
+#include "cds/archiveBuilder.hpp"
+#include "memory/allocation.hpp"
+#include "memory/allStatic.hpp"
+#include "oops/oopsHierarchy.hpp"
+#include "utilities/globalDefinitions.hpp"
+#include "utilities/growableArray.hpp"
+
+class ArchiveHeapInfo;
+class DumpRegion;
+class FileMapInfo;
+class outputStream;
+
+// Write detailed info to a mapfile to analyze contents of the AOT cache/CDS archive.
+// -Xlog:aot+map* can be used both when creating an AOT cache, or when using an AOT cache.
+//
+// Creating cache:
+//     java -XX:AOTCacheOutput=app.aot -Xlog:aot+map*=trace -cp app.jar App
+//
+// Using cache:
+//     java -XX:AOTCache=app.aot -Xlog:aot+map*=trace -cp app.jar App
+//
+// You can also print the map of a cache without executing the application by using the
+// --version flag:
+//     java -XX:AOTCache=app.aot -Xlog:aot+map*=trace --version
+//
+// Because the output can be large, it's best to save it to a file
+//     java -XX:AOTCache=app.aot -Xlog:aot+map*=trace:file=aot.map:none:filesize=0 --version
```

Comment:
why `filesize=0`?

Reply:
The map file is very large and by default UL will break it up into multiple files, each 20MB. `filesize=0` suppresses this behavior and keeps the file in a single piece. I added comments about this.